app/org/sagebionetworks/bridge/dynamodb/DynamoTaskDao.java

```diff
@@ -72,19 +71,16 @@ public boolean taskRunHasNotOccurred(String healthCode, String runKey) {
         hashKey.setHealthCode(healthCode);
         hashKey.setRunKey(runKey);
         
-        DynamoDBQueryExpression<DynamoTask> query = new DynamoDBQueryExpression<DynamoTask>()
-            .withHashKeyValues(hashKey);
-
-        return (mapper.count(DynamoTask.class, query) == 0);
+        RangeKeyCondition rangeKeyCondition = new RangeKeyCondition("runKey").eq(runKey);
+        int count = index.queryKeyCount("healthCode", healthCode, rangeKeyCondition);
+        return (count == 0);
     }
     
     /** {@inheritDoc} */
     @Override
     public void saveTasks(String healthCode, List<Task> tasks) {
         if (!tasks.isEmpty()) {
-            for (Task task : tasks) {
-                task.setHealthCode(healthCode);
-            }
+            // Health code is (now) set during construction in the scheduler.
             List<FailedBatch> failures = mapper.batchSave(tasks);
             BridgeUtils.ifFailuresThrowException(failures);
         }
```

Comment:
This method now no longer uses healthCode. Do we want to remove it from the method signature?